packages/core/src/exposed-thing.ts

```diff
@@ -316,18 +141,30 @@ export default class ExposedThing extends TD.Thing implements WoT.ExposedThing {
     }
 
     extendInteractions(): void {
+        // property defaults
         for (const propertyName in this.properties) {
-            const newProp = Helpers.extend(this.properties[propertyName], new ExposedThingProperty(propertyName, this));
-            this.properties[propertyName] = newProp;
+            const prop: TD.ThingProperty = this.properties[propertyName];
+            if (prop.readOnly === undefined || typeof prop.readOnly !== "boolean") {
+                prop.readOnly = false;
+            }
+            if (prop.writeOnly === undefined || typeof prop.writeOnly !== "boolean") {
+                prop.writeOnly = false;
+            }
+            if (prop.observable === undefined || typeof prop.observable !== "boolean") {
+                prop.observable = false;
+            }
         }
+        // action defaults
         for (const actionName in this.actions) {
-            const newAction = Helpers.extend(this.actions[actionName], new ExposedThingAction(actionName, this));
-            this.actions[actionName] = newAction;
-        }
-        for (const eventName in this.events) {
-            const newEvent = Helpers.extend(this.events[eventName], new ExposedThingEvent(eventName, this));
-            this.events[eventName] = newEvent;
+            const act: TD.ThingAction = this.actions[actionName];
+            if (act.safe === undefined || typeof act.safe !== "boolean") {
+                act.safe = false;
+            }
+            if (act.idempotent === undefined || typeof act.idempotent !== "boolean") {
+                act.idempotent = false;
+            }
         }
+        // event defaults -> none
```

Comment:
Shouldn't we handle defaults in `td-parse` ?

Reply:
> To be honest... I think we can safely remove them.. but wasn't sure what to do.
Shall we remove it.. and look into it again IF we run into any issue?

+1

Reply:
@relu91 please have a look at the additional commit